admin/src/portr_admin/services/user.py

```diff
@@ -35,16 +61,7 @@ async def get_or_create_user_from_github(code: str):
     if not github_user["email"]:
         raise EmailFetchError("No verified email found")
 
-    is_superuser = await User.filter().count() == 0
-
-    if not is_superuser:
-        if not await TeamUser.filter(user__email=github_user["email"]).exists():
-            raise UserNotFoundError("User not part of any team")
-
-    user, _ = await User.get_or_create(
-        email=github_user["email"],
-        defaults={"is_superuser": is_superuser},
-    )
+    user = await get_or_create_user(github_user["email"])
```

Comment:
Now that I think of it, `get_or_create_user` should do exactly what it says. The check should be on the existing user, whether the user can log in or not.

Reply:
Okay - sure, I can agree with that.  You mean that the business logic for handling "Is this user authenticated & can log in" vs "I want to get a user for this email whether or not it exists" should be separate.

However, that would require changing things a bit more than I had here and would be closer to a refactor of the logic, right?

I can approach that, but I was trying to change as little as possible to fix the bug.

Reply:
Perhaps a separate function in the `user` service `can_log_in` which has the business logic separately?

Reply:
Making that change and moving that check outside of both `get_or_create_user_from_github` and this new function `get_or_create_user` would change up the logic:

A `GithubUser` will be created even if a user can't technically log in.  Maybe that's ok?

Reply:
Made that change, let me know what you think.

Reply:
> A GithubUser will be created even if a user can't technically log in. Maybe that's ok?

We don't want to create users if they are not added to any teams.

Reply:
So there should be two checks
- If the user doesn't exist, can the user be created?
- If the user exists, can the user log in?

Reply:
A user won't be created unless they're the first user.  A user that already exists (as in they were invited) but hasn't been associated with GitHub would be associated with GitHub.

This is an edge case for a user that was invited but then removed from the team before they logged in the first time.

It's a `GitHubUser` link being created in an edge case - not a `User`.  So what you described in your comment for desired business logic is the current behavior.

Reply:
Linking github user is fine